### PR TITLE
New consistent header

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #273 

Adds new dark header to snapcraft.io, removed Tutorial link, added My snaps (linking to dashboard).

Designs:
https://github.com/canonicalltd/snappy-design-squad/issues/387 (header)
https://github.com/canonicalltd/snapcraft-design/issues/287 (footer)

### QA

- ./run
- Go to home page
- It should have new dark navigation
- Go to snap details page
- It should have new dark navigation

screen shot 2018-03-15 at 13 56 41

### CODE REVIEW

**Comment by bartaz** [comment]
@Greggless @lyubomir-popov Do we want/need external link icons on anything in header and footer? Designs didn't have them, but I kept them where we have them now.

I can remove them if needed.

**Comment by lyubomir-popov** [comment]
Not sure if we should have those external link icon, @Greggless what do you think?

@bartaz One minor thing - spacing in the footer seems off - I think it's the icons that throw things off. Can the last two lines be as close together as the lines above (24px line height)?

If not, can the space be the same as the one empty line between the copyright and the last two?

**Comment by bartaz** [comment]
@lyubomir-popov You are right, Vanilla "footer links" pattern (that adds the dot) has some own spacing which makes these look off. I'll try to work around that. Probably it would be easier to keep it separate (having default Vanilla spacing between them).

**Comment by bartaz** [comment]
@Greggless @lyubomir-popov Demo updated.
- removed external links from header
- improved spacing in footer

http://snapcraft.io-pr-345.run.demo.haus/

**Comment by lyubomir-popov** [comment]
@bartaz One thing about the header - on mobile the height jumps from 48 to 56px. Can it stay at 48 instead? Also, the separators are quite light for this dark background - can we

**Comment by bartaz** [comment]
Navigation border and hover colours updated: http://snapcraft.io-pr-345.run.demo.haus/

For the mobile heading size - it would be good to keep it as it is (Vanilla default), but we can override it not to have additional spacing on mobile if we want.

**Comment by lyubomir-popov** [comment]
thanks @bartaz, I guess best to consult @kwm14 on the difference in mobile height. Karl, is the change in height of the header between desktop and mobile intentional? (In vanilla, the logo gets more margin on mobile).

**Comment by kwm14** [comment]
@lyubomir-popov no there shouldn't be a difference in desktop to mobile height. I'm currently reviewing design specs to make sure they match in build. So I will add these comments to that [issue](https://github.com/ubuntudesign/vanilla-design/issues/186) so it's tracked. Thanks.

**Comment by bartaz** [comment]
@lyubomir-popov Is it ok if we keep it as it is and wait for Vanilla to fix the heading size issue (not to have style overrides in every project we have new header?).

**Comment by lyubomir-popov** [comment]
@bartaz sure

**Comment by Lukewh** [comment]
Has the dark header been signed off? I believe the visual designers were going to have a meeting regarding header colours, but I'm not sure of the outcome or if it happened? @lyubomir-popov @spencerbygraves?

**Comment by bartaz** [comment]
@Lukewh Designs come from: https://github.com/canonicalltd/snappy-design-squad/issues/387 (header) and https://github.com/canonicalltd/snapcraft-design/issues/287 (footer)

**Comment by lyubomir-popov** [comment]
@Lukewh @spencerbygraves yes, we agreed to keep the header dark, but revert the footer to light grey.

**Comment by Lukewh** [comment]
Thanks @lyubomir-popov @bartaz QA looks good too in that case :)! :+1:

**Comment by Greggless** [comment]
:shipit:

**Comment by bartaz** [comment]
Don't merge for now - to be coordinated with consistent header on other sites.

**Comment by matthewpaulthomas** [comment]
Even if this wasn’t blocked on [snappy-design-squad#392](https://github.com/CanonicalLtd/snappy-design-squad/issues/392), there are two glaring inconsistencies between the screenshot shown here and the design shown in [snapcraft-design#258](https://github.com/CanonicalLtd/snapcraft-design/issues/258) and [snappy-design-squad#387](https://github.com/CanonicalLtd/snappy-design-squad/issues/387):
- it still has a “Tutorials” link, while the design does not;
- it doesn’t link to Dashboard, while the design does.

**Comment by bartaz** [comment]
@matthewpaulthomas Yes, I don't see this in comments here so maybe it was discussed with @Greggless on slack where we made decision only to update style of header without changing the links, because we were not adding 'Sign in' to snapcraft.io yet (we need to have publisher pages ready first).

So discussing what should be in header on which site (and when it can go live) is part of clarifications needed in https://github.com/CanonicalLtd/snappy-design-squad/issues/392

**Comment by Lukewh** [comment]
@bartaz @matthewpaulthomas - it was agreed to update the navigation content at the point we're ready to go live (9th post-it down on https://github.com/ubuntudesign/meeting-notes/issues/239)

Though I do believe we should remove the Tutorials link sooner rather than later.

**Comment by bartaz** [comment]
@Lukewh @matthewpaulthomas 

As discussed on planning today:
Updated to header to remove Tutorials link. 'My snaps' in not added until we go live with publisher pages.

Could you give it a second look?

**Comment by matthewpaulthomas** [comment]
> it was agreed to update the navigation content at the point we're ready to go live (9th post-it down on ubuntudesign/meeting-notes#239)
> Though I do believe we should remove the Tutorials link sooner rather than later

> Updated to header to remove Tutorials link. 'My snaps' in not added until we go live with publisher pages.

I think it’s great that we’re removing Tutorials from the header now. But if so, then it obviously isn’t true that we can or should only “update the navigation content at the point we're ready to go live”.

That’s not surprising, because our use of “go live” here would be meaningless to any snap publisher outside Canonical. There’s a publishing UI right now, and there’ll be a publishing UI later with different design and different URLs. But those changes are pretty much irrelevant to whether it should be in the header or not. (The strongest argument against, I think, is that for now it would need an external-link icon to warn you that Dashboard’s navigation is different.)

One of our test participants said, “One thing that I still find, now that I’ve seen all of this, right, one thing that I Google the most, to this day, is where the fricking dashboard is”. The sooner we fix that, the more pain we prevent. So if we’re not blocked on making changes to the header now, and the publishing UI should be in the header later, then it should be there now too, and the best time to add it is while we’re making other changes to the header. Once our recommended publisher UI switches from Dashboard to snapcraft.io, we can change where the link points, and remove the external-link icon.

**Comment by Lukewh** [comment]
Let me rephrase:

@bartaz @matthewpaulthomas - it was agreed to add 'My snaps' and 'Login' to the navigation at the point we're ready to go live (9th post-it down on ubuntudesign/meeting-notes#239)

Though I do believe we should remove the Tutorials link sooner rather than later.

**Comment by Greggless** [comment]
I will update the [original issue](https://github.com/CanonicalLtd/snappy-design-squad/issues/387) with my work. From there we can discuss for weeks on end and create PRs based off of work that comes out of there. WOOOO 🎉 🎈 🤦‍♂️ 🔫

**Comment by matthewpaulthomas** [comment]
Sorry for the spanner in the works. I really really want to see this land. I’m just irritated that it diverges from the approved design, avoiding an easy fix, for no obvious reason.

To summarize, reasons to link to Dashboard in the header in this PR:
- It solves a problem we know users are having.
- It’s really easy to do.
- While making other big changes to the header is the best time to do it. (More noticeable, less churn.)

Arguments I’m aware of for not doing it in this PR:
- We’re not changing the header content: That’s assuming the question, and anyway, we already are.
- We’re releasing new publishing UI in a few weeks/months: Not relevant. We can change the link target once the new UI becomes more useful than the old.
- Dashboard won’t ever have the unified header: So give the link an external-link icon for now.

**Comment by Greggless** [comment]
I support [MPT's comment](https://github.com/canonical-websites/snapcraft.io/pull/345#issuecomment-373330687) we _could_ make dashboard have a consistent header if we wanted to, as described in [my comment](https://github.com/CanonicalLtd/snapcraft-design/issues/258#issuecomment-373324663) on the original issue. But that shouldn't block this from landing...

**Comment by bartaz** [comment]
@matthewpaulthomas @Greggless 

Thanks for comments and clarifications. So, to finish this stage, we just add 'My snaps (ext)' to the navigation, linking currently to dashboard.snapcraft.io/snaps? - both on snapcraft.io and build.
We leave the rest of the sites headers untouched for now with next changes being applied around 18.04 when account page will go live together with sign in on snapcraft.io?

**Comment by Greggless** [comment]
Yes please, thanks @bartaz 

# 🐦

**Comment by bartaz** [comment]
@Greggless @matthewpaulthomas I added the 'My snaps' link. Please have a look and +1:

http://snapcraft.io-pr-345.run.demo.haus/
